Automattic/wp-calypso
# Revert code that force-inserts `Settings > Newsletter` into sidebar

Related to https://github.com/Automattic/dotcom-forge/issues/8069
Complemented by https://github.com/Automattic/jetpack/pull/38232

## Proposed Changes

* Revert redundant code from #81722

## Why are these changes being made?


The original purpose of this PR was to hide the `My Mailboxes` sidebar link on P2 sites (since email isn't supported on P2 sites). However, I later realized that https://github.com/Automattic/jetpack/pull/38232 was enough to achieve this, and we didn't need any changes in Calypso.

Still, this PR removes some old logic from #81722 that was introduced as a temporary measure until https://github.com/Automattic/jetpack/pull/32992 landed in prod. Well, it's landed now, so it makes sense to remove the temporary code from Calypso. 

Since @mashikag is on leave, I'm tagging @yansern and @simison for review here (since they reviewed #81722)

## Testing Instructions



1. Navigate to the local Calypso home page for an Atomic site with the `Admin Interface Style` setting is set to `Classic style`
2. Navigate to the same page on wordpress.com
3. Compare the links sidebar on the two pages. The local version should not have the `Settings > Newsletter` link, but other than that, they should be identical
